eng/common/pipelines/templates/steps/cache-ps-modules.yml

```
@@ -0,0 +1,10 @@
+steps:
+- pwsh: |
+    . ./eng/common/scripts/Helpers/PSModule-Helpers.ps1
+    Write-Host "##vso[task.setvariable variable=CachedPSModulePath]$global:CurrentUserModulePath"
```

I'm surprised this works and questioning whether it does as expected. Because you're putting a scoped variable in a string, you need to fully contain it:
```suggestion
    Write-Host "##vso[task.setvariable variable=CachedPSModulePath]${global:CurrentUserModulePath}"
```
This didn't used to work in PowerShell. Perhaps newer versions fixed this, but to be safe I'd use the curly braces.

---

Thanks for the suggestion I will definitely verify that it works as I expect but I do believe PS has added support for scoped variables inside strings without the need for the curly braces now.